evennia/contrib/base_systems/character_creator/example_menu.py

```diff
@@ -0,0 +1,444 @@
+
+    text = dedent("""\
+        |wOption Categories|n
+
+        Some character attributes are part of the same mechanic or decision,
+        but need to be divided up into sub-categories. Character appearance
+        details are an example of where this can be useful.
+        """)
+
+    help = "Some helpful extra information on what's affected by these choices works well here."
+    options = []
+
+    # just like for informational categories, build the options off of a dictionary to make it easier to manage
+    for category in _APPEARANCE_DICT.keys():
+        options.append({"desc": f"Choose your |c{category}|n", "goto": ("menunode_category_options", { "category": category })})
+
+    # since this node goes in and out of sub-nodes, you need an option to proceed to the next step
+    options.append({"key": ("(Next)", "next", "n"), "desc": "Continue to the next step.", "goto": "menunode_multi_choice"})
+    # once past the first decision, it's also a good idea to include a "back to previous step" option
+    options.append({"key": ("(Back)", "back", "b"), "desc": "Go back to the previous step", "goto": "menunode_info_base"})
+    return (text, help), options
+
+def menunode_category_options(caller, raw_string, category=None, **kwargs):
+    """Choosing an option within the categories."""
+    if not category:
+        # this shouldn't have happened, so quit and retry
+        return "Something went wrong. Please try again."
+
+    # for mechanics-related choices, you can combine this with the
+    # informational options approach to give specific info
+    text = f"Choose your {category}:"
+    help = f"This will define your {category}."
+
+    options = []
+    # build the list of options from the right category of your dictionary
+    for option in _APPEARANCE_DICT[category]:
+        options.append({"desc": option, "goto": (_set_category_opt, { "category": category, "value": option})})
+    # always include a "back" option in case they aren't ready to pick yet
+    options.append({"key": ("(Back)", "back", "b"), "desc": f"Don't change {category}", "goto": "menunode_categories"})
+    return (text, help), options
+
+def _set_category_opt(caller, raw_string, category, value, **kwargs):
+    """Set the option for a category"""
+    
+    # this is where you would put any more complex code involved in setting the option,
+    # but we're just doing simple attributes
+    caller.new_char.attributes.add(category, value)
+
+    # go back to the base node for the categories choice to pick another
+    return "menunode_categories"
+
+
+#########################################################
+#                  Multiple Choice
+#########################################################
+
+# it's not as important to make this a separate list, but like all the others,
+# it's easier to read and to update if you do!
+_SKILL_OPTIONS = [ "alchemy", "archery", "blacksmithing", "brawling", "dancing", "fencing", "pottery", "tailoring", "weaving" ]
```

Comment:
This works fine, but it's worth to note that `evennia.utils.evmenu.list_node` decorator is specifically there to auto-generate a multiple-choice menu node. So it could be worth to point to that since it would lead to a lot less code.